src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -189,7 +189,7 @@ extends AbstractSeq[T] with IndexedSeq[T] with Serializable {
       import numAsIntegral._
       if (isEmpty) num fromInt 0
       else if (numRangeElements == 1) head
-      else ((num fromInt numRangeElements) * (head + last) / (num fromInt 2))
+      else (num fromInt numRangeElements) * ((head + last) / (num fromInt 2))
```

Comment:
Actually, your new formula is wrong for integer ranges with an odd step and an even number of elements. For example:
`2 5 8 11` (step = 3, numElements = 4)
Your formula gives: `4 * ((2 + 11) / 2) == 4 * (13 / 2) == 4 * 6 == 24` instead of 26. This is because `13 / 2` is not exact.

The old formula did not have this problem, because the numElements is even, and hence the division is exact. Similarly, if step is even, then `head + last` is necessarily even; and if numElements is odd, the same happens.

I suggest
```scala
...
else if (numRangeElements % 2 == 0) (num fromInt (numRangeElements / 2)) * (head + last)
else (num fromInt numRangeElements) * ((head + last) / (num fromInt 2))
```
* If `(head + last)` overflows, then the total result would overflow anyway, because we have at least 2 elements; so we can assume `(head + last)` does not overflow.
* If `numRangeElements` is even, the first formula must be exact. Only the last multiplication can overflow, which means there's no overflow unless the final result overflows anyway.
* If it is odd, then we know that `(head + last)` must be even, and hence the second formula must be exact. Again, only the last multiplication can overflow.